cmd/server/main.go

```diff
@@ -33,15 +36,31 @@ func main() {
 	if err != nil {
 		logger.Fatalln(err)
 	}
+
+	middleware := grpc_middleware.ChainUnaryServer(
+		grpc_validator.UnaryServerInterceptor(),                     // validation middleware
+		grpc_logrus.UnaryServerInterceptor(logrus.NewEntry(logger)), // logging middleware
+	)
+	// add authorization middleware if the pdp address is provided
+	if PDPAddr != "" {
+		authorizer := toolkit_auth.Authorizer{
```

Comment:
it'd be awesome if this whole `authorizer` could be done in a single line of code:
```
authorizer, err := toolkit_auth.NewDefaultAuthorizer(PDPAddr)
```
Maybe it could even do a connection check or something and fail early if the auth service is down. Not sure if that's a good pattern or not, though..

Reply:
> it'd be awesome if this whole authorizer could be done in a single line of code:

I'm conflicted about that. On one hand, it would provide a small amount of convenience to the developer. On the other hand, it would hide implementation details from them. If someone is DIYing their own application based on this example, I would rather not hide the functional options pattern behind a `New` function.

> Maybe it could even do a connection check or something and fail early if the auth service is down. Not sure if that's a good pattern or not, though..

At the very least, a `WARN` message would be good. That's something I can add to the toolkit.

Reply:
This is going to be the overwhelming majority of configurations for auth. As it stands now, the consumer of this package has to know about a bunch of different things to wire it together:
1. I have to import and instantiate a `toolkit_auth.Authorizer` struct
2. pass into one of those fields a `PDPAddr`
3. learn about and create something called a `Builder`
4. know what a `JWT` is, know to pass in a `nil` argument
5. know what a `WithRequest` thing means, and to include it in my thing
6. know what a `Handler` is, and import and instantiate the default one
7. know to pass this into a `grpc_auth.UnaryServerInterceptor` function by calling `.AuthFunc()`

Odds are they don't care enough about auth to learn all this -- they just need to add the thing to their thing. They're just going to find someone else's implementation and copy/paste it into their own. I'd be more comfortable with them copy/pasting a single line than all of this. If they need to have custom functionality, they can follow the source to the `NewDefaultAuthorizer` func and see all of the moving parts.

Reply:
Sounds good! I'll add this functionality to the toolkit.